### PR TITLE
[Frontend] Updates to CUDA Quantum

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Description of the Change:**
* Enables CSWAP
* Converts QuakeValue to jax.numpy.array for `qml.state`.
* Fixes bitstrings obtained from `qml.sample`
* Removes device map / uses device names
* Enable other simulator devices

[sc-57343]

### CODE REVIEW

**Review by josh146 — APPROVED** [review]
Nice, thanks @erick-xanadu!

File: frontend/catalyst/cuda/__init__.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class BaseCudaInstructionSet(qml.QubitDevice):
     """Base instruction set for CUDA-Quantum devices"""
 
     # TODO: Once 0.35 is released, remove -dev suffix.
-    pennylane_requires = "0.35.0-dev"
+    pennylane_requires = "0.34"
```
Comment (dime10) [inline frontend/catalyst/cuda/__init__.py:45]: We may even want to set this to "0.35" now, unless this is actually checked anywhere?

File: setup.py [hunk 2]
```diff
@@ -51,7 +51,11 @@
 one_compiler_per_distribution = pl_version == ">=0.32,<=0.34"
```
Comment (dime10) [inline setup.py:51]: Didn't we remove this 👀

**Comment by dime10** [comment]
Oh, can we fix this warning as well? 

> frontend/catalyst/cuda/catalyst_to_cuda_interpreter.py:850:12: W0632: Possible unbalanced tuple unpacking with sequence defined at line 359 of catalyst.jax_tracer: left side has 4 labels, right side has 2 values (unbalanced-tuple-unpacking)

File: setup.py (hunk 2 above)
Comment (erick-xanadu) [reply setup.py:51]: It is in the other PR.

File: frontend/catalyst/cuda/__init__.py (hunk 1 above)
Comment (erick-xanadu) [reply frontend/catalyst/cuda/__init__.py]: I just changed it to ">=0.34"

**Review by maliasadi — APPROVED** [review]
:tada:
